hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/WritableScheduleStoreImpl.java

```diff
@@ -0,0 +1,139 @@
+    private static final String SCHEDULE_NULL_FOR_DELETE_MESSAGE =
+            "Request to delete null schedule ID cannot be fulfilled.";
+    private static final String SCHEDULE_MISSING_FOR_DELETE_MESSAGE =
+            "Schedule to be deleted, %1$s, not found in state.";
+    private final WritableKVState<ScheduleID, Schedule> schedulesByIdMutable;
+    private final WritableKVState<String, List<Schedule>> schedulesByEqualityMutable;
+    private final WritableKVState<Long, List<Schedule>> schedulesByExpirationMutable;
+
+    /**
+     * Create a new {@link WritableScheduleStoreImpl} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableScheduleStoreImpl(@NonNull final WritableStates states) {
+        super(states);
+        schedulesByIdMutable = states.get(ScheduleServiceImpl.SCHEDULES_BY_ID_KEY);
+        schedulesByEqualityMutable = states.get(ScheduleServiceImpl.SCHEDULES_BY_EQUALITY_KEY);
+        schedulesByExpirationMutable = states.get(ScheduleServiceImpl.SCHEDULES_BY_EXPIRY_SEC_KEY);
+    }
+
+    /**
+     * Delete a given schedule from this state.
+     * Given the ID of a schedule and a consensus time, delete that ID from this state as of the
+     * consensus time {@link Instant} provided.
+     * @param scheduleToDelete The ID of a schedule to be deleted.
+     * @param consensusTime The current consensus time
+     * @return the {@link Schedule} marked as deleted.
+     * @throws IllegalStateException if the {@link ScheduleID} to be deleted is not present in this state,
+     *     or the ID value has a mismatched realm or shard for this node.
+     */
+    @SuppressWarnings("DataFlowIssue")
+    @Override
+    @NonNull
+    public Schedule delete(@Nullable final ScheduleID scheduleToDelete, @NonNull final Instant consensusTime) {
+        Objects.requireNonNull(consensusTime, "Null consensusTime provided to schedule delete, cannot proceed.");
+        if (scheduleToDelete != null) {
+            final Schedule schedule = schedulesByIdMutable.getForModify(scheduleToDelete);
+            if (schedule != null) {
+                final Schedule deletedSchedule = markDeleted(schedule, consensusTime);
+                schedulesByIdMutable.put(scheduleToDelete, deletedSchedule);
+                return schedulesByIdMutable.get(scheduleToDelete);
+            } else {
+                throw new IllegalStateException(SCHEDULE_MISSING_FOR_DELETE_MESSAGE.formatted(scheduleToDelete));
+            }
+        } else {
+            throw new IllegalStateException(SCHEDULE_NULL_FOR_DELETE_MESSAGE);
+        }
+    }
+
+    @Override
+    public Schedule getForModify(@Nullable final ScheduleID idToFind) {
+        final Schedule result;
+        if (idToFind != null) {
+            result = schedulesByIdMutable.getForModify(idToFind);
+        } else {
+            result = null;
+        }
+        return result;
```

Comment:
Not a big deal, but could be reduced to a ternary statement to save a couple lines if wanted. I'm fine either way

```java
return (idToFind != null) ? schedulesByIdMutable.getForModify(idToFind) : null;
```